azurerm/internal/services/compute/ssh_keys.go

```diff
@@ -155,3 +161,33 @@ func ValidateSSHKey(i interface{}, k string) (warnings []string, errors []error)
 
 	return warnings, errors
 }
+
+func SSHKeyDiffSuppress(_, old, new string, _ *schema.ResourceData) bool {
+	oldNormalised, err := azure.NormaliseSSHKey(old)
+	if err != nil {
+		return false
+	}
+
+	newNormalised, err := azure.NormaliseSSHKey(new)
+	if err != nil {
+		return false
+	}
+
+	if *oldNormalised == *newNormalised {
+		return true
+	}
+
+	return false
+}
+
+func SSHKeySchemaHash(v interface{}) int {
+	var buf bytes.Buffer
+
+	if m, ok := v.(map[string]interface{}); ok {
+		normalisedKey, _ := azure.NormaliseSSHKey(m["public_key"].(string))
```

Comment:
we should catch and log this, fwiw:

```suggestion
		normalisedKey, err := azure.NormaliseSSHKey(m["public_key"].(string))
		if err != nil {
			log.Printf("[DEBUG] error normalizing ssh key %q: %+v", m["public_key"].(string), err)
		}
		if normalisedKey != nil {
			buf.WriteString(fmt.Sprintf("%s-", *normalisedKey))
		}
```